File: code/mathematical-algorithms/armstrong_numbers/armstrong_numbers.c
#include <stdio.h>
int main()
{
    int number, originalNumber, remainder, result = 0;

    printf("Enter a three digit integer: ");
    scanf("%d", &number);

    originalNumber = number;

    while (originalNumber != 0)
    {
        remainder = originalNumber%10;
        result += remainder*remainder*remainder;
        originalNumber /= 10;
    }

    if(result == number)
        printf("%d is an Armstrong number.",number);
    else
        printf("%d is not an Armstrong number.",number);

    return 0;
}

File: code/mathematical-algorithms/armstrong_numbers/ArmstrongNumbers.java
import java.util.Scanner;

public class ArmstrongNumbers {
	public static void main(String args[]) {
		Scanner in = new Scanner(System.in);

		System.out.println("Enter a 3-digit number : ");
		int number = in.nextInt();
		int a = number % 10;
		int b = (number / 10) % 10;
		int c = number / 100;

		if ((a*a*a + b*b*b + c*c*c) == number) {
			System.out.println(number + " is a armstrong number.");
		}
		else {
			System.out.println(number + " is not a armstrong number.");
		}
		in.close();
	}
}